src/asm/macro.cpp

```diff
@@ -80,12 +80,6 @@ void macro_UseNewArgs(struct MacroArgs *args)
 	macroArgs = args;
 }
 
-void macro_FreeArgs(struct MacroArgs *args)
```

Comment:
How come this is never called?

Reply:
I think it's because symbols could be referring to their contents. I tried calling this in a few likely places and got double-frees. If we're not going to actually deal with #709 then I don't think we need this function to pre-emptively exist.

Reply:
Symbols can't reference the contents of macro args directly, since they are always expanded and not used any other way (???)

The place that this should be called is in `yywrap`, specifically:

```diff
 	// Restore args if a macro (not REPT) saved them
-	if (context->fileInfo->type == NODE_MACRO)
+	if (context->fileInfo->type == NODE_MACRO) {
+		macro_FreeArgs(macro_GetCurrentArgs());
 		macro_UseNewArgs(contextStack->macroArgs);
```

*Please* tell me that doesn't double-free.

Reply:
Looks like it works! Thank you :D

(Also, after working on libplum recently, it's nice to see `context->fileInfo->type` and not `context -> fileInfo -> type`. :P )

Reply:
> (Also, after working on libplum recently, it's nice to see `context->fileInfo->type` and not `context -> fileInfo -> type`. :P )

BOO!

Reply:
This is *highly poggers*, if you'll allow me the term.

Also, nobody was expecting a *coverage* PR to fix a memory leak, but hey, here we are! :D

Reply:
A memory leak *and* a minor arithmetic bug!